src/test/java/com/example/bookshop/BookingControllerTests.java

```diff
@@ -0,0 +1,69 @@
+package com.example.bookshop;
+
+import com.example.bookshop.controller.*;
+import com.example.bookshop.entity.Booking;
+import com.example.bookshop.entity.Status;
+import com.example.bookshop.entity.User;
+import com.example.bookshop.pojos.ProductObj;
+import com.example.bookshop.pojos.RoleObj;
+import com.example.bookshop.pojos.StatusObj;
+import com.example.bookshop.pojos.UserObj;
+import com.example.bookshop.pojos.enums.RoleType;
+import com.example.bookshop.pojos.enums.StatusType;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.event.annotation.AfterTestMethod;
+import org.springframework.test.context.event.annotation.BeforeTestMethod;
+
+import java.util.ArrayList;
+import java.util.Random;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
+
+@SpringBootTest()
+class BookingControllerTests {
+    @Autowired
+    private BookingController bookingController;
+    @Autowired
+    private UserController userController;
+
+    UserObj userObj;
+
+
+    @Test
+    public void addNewBooksObj_Status200() {
```

Comment:
I don't see status 200 is actually tested here. That could be nice test for Service layer (where some logic happens)
If you test controlller, make sense to actually send API calls and verify status codes, headers, responces
https://www.baeldung.com/spring-boot-testing

Reply:
got it!